### DISCUSSION TITLE
Editing Singleplayer & Multiplayer Savegames

### DISCUSSION THREAD

**Discussion by strich** [diasurgical/DevilutionX#2875] [Ideas]
With classic Diablo, there are various ancient editors available for editing heros and savegames. AFAIK none of these work with DevilutionX? Would it be possible to make savegames or open for editing?

In addition to this, it would be very useful to allow migrating between singleplayer and multiplayer.

**Comment by StephenCWills** [comment]
FYI, this discussion seems relevant. #2548

**Comment by StephenCWills** [comment]
I don't know about the compatibility of savegame editors, but DevilutionX saves are compatible with vanilla Diablo and Hellfire. As for supporting any external tools for editing savegames, I think we don't really need to since we have debug commands built into the game itself. Is there something specific you're trying to accomplish?

**Reply by strich** [reply]
Basically: I've put a bunch of time into a singleplayer character and would like to replicate it to multiplayer so I can play it there. I'm happy to recreate a character if needed - I wasn't aware there are debug commands available. Is there doco?

**Reply by StephenCWills** [reply]
You need to use a debug build, but you can type `help` into the in-game chat box to get a list of commands. You can also reference the code.

https://github.com/diasurgical/devilutionX/blob/8a71b9ded31ba392f27e1710357b972a2f00ef9c/Source/debug.cpp#L617-L644
